File: Stage32/C/Problem13913.c
#include <stdio.h>

#define MAX 100001

int dist[MAX];
int prev[MAX];
int queue[MAX];

void bfs(int start, int target) {
    for (int i = 0; i < MAX; i++) dist[i] = -1;

    int front = 0, rear = 0;
    queue[rear++] = start;
    dist[start] = 0;

    while (front < rear) {
        int cur = queue[front++];

        if (cur == target) return;

        int next[3] = {cur - 1, cur + 1, cur * 2};

        for (int i = 0; i < 3; i++) {
            int nx = next[i];

            if (nx >= 0 && nx < MAX && dist[nx] == -1) {
                dist[nx] = dist[cur] + 1;
                prev[nx] = cur;
                queue[rear++] = nx;
            }
        }
    }
}

void printPath(int start, int target) {
    int path[MAX];
    int idx = 0;
    int cur = target;

    while (cur != start) {
        path[idx++] = cur;
        cur = prev[cur];
    }
    path[idx++] = start;

    for (int i = idx - 1; i >= 0; i--) {
        printf("%d ", path[i]);
    }
}

int main() {
    int N, K;
    scanf("%d %d", &N, &K);

    bfs(N, K);

    printf("%d\n", dist[K]);
    printPath(N, K);

    return 0;
}

File: Stage32/Java/Problem13913.java
package Stage32.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Stack;
import java.util.StringTokenizer;

public class Problem13913 {
    static final int MAX = 100001;
    static int[] dist = new int[MAX];
    static int[] prev = new int[MAX];

    static void bfs(int start, int target) {
        Arrays.fill(dist, -1);

        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        dist[start] = 0;

        while (!q.isEmpty()) {
            int cur = q.poll();

            if (cur == target) return;

            int[] next = {cur - 1, cur + 1, cur * 2};

            for (int nx : next) {
                if (nx >= 0 && nx < MAX && dist[nx] == -1) {
                    dist[nx] = dist[cur] + 1;
                    prev[nx] = cur;
                    q.offer(nx);
                }
            }
        }
    }

    static void printPath(int start, int target) {
        Stack<Integer> stack = new Stack<>();
        int cur = target;

        while (cur != start) {
            stack.push(cur);
            cur = prev[cur];
        }
        stack.push(start);

        while (!stack.isEmpty()) {
            System.out.print(stack.pop() + " ");
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        bfs(N, K);

        System.out.println(dist[K]);
        printPath(N, K);
    }
}
